crates/biome_js_parser/src/lexer/mod.rs

```diff
@@ -1727,6 +1738,82 @@ impl<'src> JsLexer<'src> {
         }
     }
 
+    /// Attempts to lex a Glimmer template block `<template>...</template>`.
+    /// Returns Some(GLIMMER_TEMPLATE) if a template was found, None otherwise.
+    fn try_lex_glimmer_template(&mut self) -> Option<JsSyntaxKind> {
+        // We're currently at '<', check if followed by 'template'
+        // Peek ahead without consuming
+        let start = self.position;
+
+        // Check for "<template"
+        if self.peek_byte() != Some(b't') {
+            return None;
+        }
+
+        // Check if we have enough bytes for "<template>"
+        let template_bytes = b"template>";
+        let end_pos = self.position + 1 + template_bytes.len();
+        if end_pos > self.source.len() {
+            return None;
+        }
+        if &self.source.as_bytes()[self.position + 1..end_pos] != template_bytes {
+            return None;
```

Comment:
When `try_lex_glimmer_template` returns `None`, does the lexer back up to the start properly?

The error tests need to be comprehensive enough to answer this question.